Frameworks/UIKit/UIViewController.mm

```diff
@@ -1336,28 +1349,23 @@ - (void)_addToTop:(NSNumber*)animatedValue {
 
     if ([self parentViewController] != nil) {
         if ([self modalPresentationStyle] == UIModalPresentationPopover) {
-            if (priv->_presentationController != nil) {
-                [priv->_presentationController release];
-            }
-
-            priv->_presentationController = priv->_popoverPresentationController;
-
             [priv->_popoverPresentationController _prepareForPresentation];
 
             displayPopover = ![self _hidesParent];
 
             if (displayPopover) {
-                priv->_popoverPresentationController->_managesPresentation = TRUE;
+                priv->_popoverPresentationController->_managesPresentation = YES;
```

Comment:
>_popoverPresentationController->_managesPresentation [](start = 22, length = 52)

it's unfortunate that we have to reach into _popoverPresentationController's internal state in so many places.  Is there any way to further encapsulate this block of logic?  Most of it seems like it should be handled by the UIPopoverPresentationController.